ipie/analysis/extraction.py

```diff
@@ -68,6 +68,49 @@ def extract_observable(filename, name="energy", block_idx=1):
         return walker_averaged.reshape((nsamp,) + tuple(obs_info["shape"]))
 
 
+def extract_hdf5_data_complex(filename, block_idx=1):
+    shapes = {}
+    with h5py.File(filename, "r") as fh5:
+        keys = fh5[f"block_size_{block_idx}/data/"].keys()
+        shape_keys = fh5[f"block_size_{block_idx}/shape/"].keys()
+        data = numpy.concatenate([fh5[f"block_size_{block_idx}/data/{d}"][:] for d in keys])
+        for k in shape_keys:
+            shapes[k] = {
+                "names": fh5[f"block_size_{block_idx}/names/{k}"][()],
+                "shape": fh5[f"block_size_{block_idx}/shape/{k}"][:],
+                "offset": fh5[f"block_size_{block_idx}/offset/{k}"][()],
+                "size": fh5[f"block_size_{block_idx}/size/{k}"][()],
+                "scalar": bool(fh5[f"block_size_{block_idx}/scalar/{k}"][()]),
+                "num_walker_props": fh5[f"block_size_{block_idx}/num_walker_props"][()],
+                "walker_header": fh5[f"block_size_{block_idx}/walker_prop_header"][()],
+            }
+        size_keys = fh5[f"block_size_{block_idx}/max_block"].keys()
+        max_block = sum(fh5[f"block_size_{block_idx}/max_block/{d}"][()] for d in size_keys)
+
+    return data[: max_block + 1], shapes
+
+
+def extract_observable_complex(filename, name="energy", block_idx=1):
```

Comment:
Shouldn't the name "complex" be called "no imporatnace sampling"?

Reply:
renamed after moving